parser/parser_system1.c

```diff
@@ -1118,6 +1118,36 @@ static int do_action_for_drivers(struct json_key_action *jka, char *value)
 	return 0;
 }
 
+static qos_policy_t parse_qos_policy(char *value, size_t len)
+{
+	if (pv_str_matches(value, len, "guaranteed", strlen("guaranteed")))
+		return QOS_GUARANTEED;
+	else if (pv_str_matches(value, len, "besteffort", strlen("besteffort")))
+		return QOS_BESTEFFORT;
+	else if (pv_str_matches(value, len, "burstable", strlen("burstable")))
+		return QOS_BURSTABLE;
+
+	pv_log(ERROR, "invalid qos policy '%s'", value);
+	return QOS_NONE;
```

Comment:
is QOS_NONE the same as QOS_UNKNOWN?

Reply:
also, bear in mind that QOS_NONE won't fail in this case

Reply:
QOS_NONE is unset, while UNKNOWN is not parsed (yet).